src/workerd/api/global-scope.c++

```diff
@@ -459,8 +459,28 @@ jsg::Promise<void> ServiceWorkerGlobalScope::test(
   return testHandler(lock, jsg::alloc<TestController>(), eh.env.addRef(lock), eh.getCtx());
 }
 
+kj::Promise<void> ServiceWorkerGlobalScope::eventTimeoutPromise(uint32_t timeout) {
```

Comment:
non-blocking nit: a comment in here explaining that this promise is expected to be dropped in the typical case in order to cancel the timeout would be helpful for others coming into the code later.